misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -12,21 +12,107 @@ export const pickingReducer = ({ draftState, action }) => {
       return reduceOnUpdateQtyPicked(draftState, action.payload);
     }
 
+    case types.UPDATE_ALT_PICKING_STEP_QTY: {
+      return reduceOnAlternatePickingQty(draftState, action.payload);
+    }
+
     default: {
       return draftState;
     }
   }
 };
 
+const reduceOnAlternatePickingQty = (draftState, payload) => {
+  const { wfProcessId, activityId, lineId, stepId, altStepId, scannedHUBarcode, qtyPicked, qtyRejectedReasonCode } =
+    payload;
+
+  const draftWFProcess = draftState[wfProcessId];
+  const draftAltStep =
+    draftWFProcess.activities[activityId].dataStored.lines[lineId].steps[stepId].altSteps.genSteps[altStepId];
+  draftAltStep.scannedHUBarcode = scannedHUBarcode;
+  draftAltStep.qtyPicked = qtyPicked;
+  draftAltStep.qtyRejectedReasonCode = qtyRejectedReasonCode;
+
+  return draftState;
+};
+
+const generateAlternativeSteps = ({ draftState, wfProcessId, activityId, lineId, stepId, qtyRejected }) => {
+  const draftDataStored = draftState[wfProcessId].activities[activityId].dataStored;
+  const draftDataStoredOrig = original(draftDataStored);
+  const draftStep = draftState[wfProcessId].activities[activityId].dataStored.lines[lineId].steps[stepId];
+  const { pickFromAlternatives } = draftDataStoredOrig;
+  let remainingQtyRejected;
+
+  for (let idx = 0; idx < pickFromAlternatives.length; idx++) {
+    if (pickFromAlternatives[idx].qtyAvailable >= qtyRejected) {
+      draftDataStored.pickFromAlternatives[idx].qtyAvailable = pickFromAlternatives[idx].qtyAvailable - qtyRejected;
+      draftStep.altSteps.genSteps[pickFromAlternatives[idx].id] = {
+        id: pickFromAlternatives[idx].id,
+        locatorName: pickFromAlternatives[idx].locatorName,
+        huBarcode: pickFromAlternatives[idx].huBarcode,
+        uom: pickFromAlternatives[idx].uom,
+        qtyAvailable: qtyRejected,
+        qtyPicked: 0,
+      };
+      break;
+    } else {
+      remainingQtyRejected = qtyRejected - pickFromAlternatives[idx].qtyAvailable;
+      draftDataStored.pickFromAlternatives[idx].qtyAvailable = 0;
+      draftStep.altSteps.genSteps[pickFromAlternatives[idx].id] = {
+        id: pickFromAlternatives[idx].id,
+        locatorName: pickFromAlternatives[idx].locatorName,
+        huBarcode: pickFromAlternatives[idx].huBarcode,
+        uom: pickFromAlternatives[idx].uom,
+        qtyAvailable: pickFromAlternatives[idx].qtyAvailable,
+        qtyPicked: 0,
+      };
+      if (remainingQtyRejected > 0) {
+        generateAlternativeSteps({
```

Comment:
pls avoid recursive algorithms by all means.
in this case, you can have a variable with `qtyToAlocateRemaining`.